[CombinerHelper]: Use undef for handling divisors of one

AZero13 opened the pull request:
I need to figure out if these changes are regressions and how to mitigate them, but they do follow the logic in TargetLowering now.

arsenm commented on llvm/lib/CodeGen/GlobalISel/CombinerHelper.cpp:
```diff
@@ -5084,23 +5086,28 @@ MachineInstr *CombinerHelper::buildUDivUsingMul(MachineInstr &MI) {
-      PostShift = magics.PostShift;
-      SelNPQ = magics.IsAdd;
+    if (Divisor.isOne()) {
+      PreShifts.push_back(MIB.buildUndef(ScalarShiftAmtTy).getReg(0));
```
Call buildUndef with the same time one time and reuse the value for each of these

paperchalice committed:
[Instrumentation] Support verifying machine function (#90931)

We need it to test isel related passes. Currently
`verifyMachineFunction` is incomplete (no LiveIntervals support), but is
enough for testing isel pass, will migrate to complete
`MachineVerifierPass` in future.
